### PR TITLE
Feature/discussion progress js

### PR DESCRIPTION
**Pull request by wykhuh**
@pigstake Here's the javascript for the discussion progress. 

I think there is a bug in the css.  I put some temp copy in each idea showing the state of the idea. When the idea `discussing: false, discussed: true`, the idea is yellow instead of grey. Can you pull down the branch to check the css and check if the feature is behaving as expected?

### CODE REVIEW

**Comment by pigstake** [comment]
@wykhuh Fixed the issue you mentioned and made some other tweaks. :eyes:

**Comment by wykhuh** [comment]
@pigstake looks good. 

@mikehmorrissey Can you look over the code? This is the most complex thing I've done in meteor so far, so a code review would be good.  When the board is closed,  there will be visual indicators if  an idea is `current/idea-discussing` or `finished/idea-discussed`.

https://trello.com/c/KUI9lDvm/136-as-a-facilitator-i-should-be-able-to-indicate-where-the-discussion-is-with-a-visual-signifier-so-that-everyone-can-follow-discus

**Comment by pigstake** [comment]
@mikehmorrissey There are some awkward moments now that I don't thin we need to address until people use these states, like what happens when you reopen a board that has ideas with ".is-discussing" or ".is-discussed". Just a heads up for you while you're looking over the code.

**Comment by pigstake** [comment]
Just rebased and added a minor bug fix

**Comment by pigstake** [comment]
@mikehmorrissey Please review this as soon as you can, want to :sheep: this ASAP

**Comment by pigstake** [comment]
@mikehmorrissey Please for the love of sheep review this :pray:

File: client/modules/idea/idea.js [hunk 1]
```diff
@@ -7,57 +7,52 @@ Template.idea.helpers({
...
-      return 'idea is-discussable';
-    } else if (canChange) {
-      return 'idea is-editable';
-    } else if (isMine) {
-      return 'idea idea--mine';
-    } else {
-      return 'idea';
-    }
-  },
+    if (canChange) { classes.push('is-editable'); }
```
Comment (mikehmorrissey) [inline client/modules/idea/idea.js]: @wykhuh :+1:

File: client/modules/idea/idea.html [hunk 2]
```diff
@@ -8,6 +8,7 @@
     <section class="idea-content">
       <p class="idea-body">
         {{ body }}
+        <br>discussing: {{ status.current }}, discussed: {{ status.finished}}
```
Comment (mikehmorrissey) [inline client/modules/idea/idea.html]: @wykhuh missing space after finished

File: client/modules/idea/idea.js [hunk 3]
```diff
@@ -7,57 +7,52 @@ Template.idea.helpers({
...
-      // (Clicking "btn-reset-idea" should move this card back to "default" state)
+      if (globals.isAdmin() && !this.finished) {
+        previousId =  Session.get('currentIdeaId');
+        Session.set('currentIdeaId', this._id);
+        currentId =  Session.get('currentIdeaId');
 
+        // if admin clicks on same idea, get rid of current idea
+        if (previousId === currentId) {
+          Ideas.update(currentId, { $set: { current: false } });
+          Session.set('currentIdeaId', null);
```
Comment (mikehmorrissey) [inline client/modules/idea/idea.js]: @wykhuh any reason this is on the Session vs the board itself?

File: client/modules/ideas/ideas.js [hunk 4]
```diff
@@ -3,11 +3,19 @@ Template.ideas.helpers({
...
+  },
+
   ideas: function() {
     if (globals.isBoardOpen()) {
       return globals.currentIdeas();
     } else {
       return globals.currentSortedIdeas();
     }
+  },
+  blah1: function(idea) {
```
Comment (mikehmorrissey) [inline client/modules/ideas/ideas.js]: @wykhuh :smiley_cat:

File: client/modules/ideas/ideas.html [hunk 5]
```diff
@@ -2,13 +2,15 @@
   <div class="ideas">
     {{ #if ideas.count }}
       {{ #each ideas }}
-        <div class="ideaWrapper">
+        <div class="ideaWrapper" data-idea-id="{{_id}}">
```
Comment (mikehmorrissey) [inline client/modules/ideas/ideas.html]: @wykhuh where is this used?

**Comment by mikehmorrissey** [comment]
@wykhuh couple comments. nothing major. address what you want and :shipit:

File: client/modules/idea/idea.js (hunk 3 above)
Comment (wykhuh) [inline client/modules/idea/idea.js]: No particular reason. I was playing around with different ways of doing things in Meteor.

File: client/modules/ideas/ideas.html (hunk 5 above)
Comment (wykhuh) [inline client/modules/ideas/ideas.html]: Left over code that I forgot to delete.

**Pull request merged**